### DISCUSSION TITLE
2D support?

### DISCUSSION THREAD

**Discussion by SteffenCzolbe** [TorchIO-project/torchio#455] [Q&A] [answered]
Hello,
Great projects you have going on here, I'm looking forward to replace my shitty nii-to-pytorch pipeline and augmentation code with this library.

However, it's unclear to me to which extend this library supports 2D images. I work with both 2d and 3d data, and using one codebase and pipline for both 2d and 3d images is very important to me. The documentation of torchio only talks about 3D images, yet the Image class seems to have some support for 2d images as well. Can you please clarify to which extend I will be able to use torchio with 2d images?

**Comment by fepegar** [comment, marked as answer]
> Hello,
Great projects you have going on here, I'm looking forward to replace my shitty nii-to-pytorch pipeline and augmentation code with this library.

:D thanks @SteffenCzolbe.

> Can you please clarify to which extend I will be able to use torchio with 2d images?

The reason why there is not a lot of 2D talk in the docs is just that the main challenges that this library solves are related to volumetric data with corresponding spatial metadata. That being said, you can see a 4D image as a generalization of a 2D image, so everything should work fine. Here's some IPython code:

```python
!curl -o image2d.jpg https://thispersondoesnotexist.com/image
import torchio as tio
image = tio.ScalarImage('image2d.jpg')
image.plot()

image  # prints: ScalarImage(shape: (3, 1024, 1024, 1); spacing: (1.00, 1.00, 1.00); orientation: LPS+; memory: 3.0 MiB; dtype: torch.ByteTensor)

add_noise = tio.RandomNoise(std=50)
noisy = add_noise(image)
noisy.plot()

elastic_transform = tio.RandomElasticDeformation(max_displacement=150)
deformed = elastic_transform(noisy)
deformed.plot()
```

These images are shown (I have rotated them 90 degrees for visualization purposes).

Loaded image:

![tmpt5lztot2 copy](https://user-images.githubusercontent.com/12688084/107230288-c9a83180-6a16-11eb-920c-40ea750e5965.png)

Noisy image:

![tmpcam74b5u copy](https://user-images.githubusercontent.com/12688084/107230286-c7de6e00-6a16-11eb-8dd9-9bd540ce5b42.png)

Deformed image:

![tmp45cj62lx copy](https://user-images.githubusercontent.com/12688084/107230271-c2812380-6a16-11eb-981a-a36d8d69bdfd.png)

Note that you'll get a different face if you run that code, as the website generates face using a GAN.

There are some more examples in the Transforms tutorial: [Data preprocessing and augmentation using TorchIO: a tutorial](https://github.com/fepegar/torchio-notebooks/blob/main/notebooks/Data_preprocessing_and_augmentation_using_TorchIO_a_tutorial.ipynb).

**Reply by fepegar** [reply]
Does that answer your question?
